// react-mapbox-gl-language/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _mapboxGlLanguage = require('@mapbox/mapbox-gl-language');

var _mapboxGlLanguage2 = _interopRequireDefault(_mapboxGlLanguage);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var MapboxLanguageControl = function (_Component) {
  _inherits(MapboxLanguageControl, _Component);

  function MapboxLanguageControl() {
    _classCallCheck(this, MapboxLanguageControl);

    return _possibleConstructorReturn(this, (MapboxLanguageControl.__proto__ || Object.getPrototypeOf(MapboxLanguageControl)).apply(this, arguments));
  }

  _createClass(MapboxLanguageControl, [{
    key: 'componentWillMount',
    value: function componentWillMount() {
      this.mapboxLanguage = new _mapboxGlLanguage2.default(this.props);
      this.context.map.addControl(this.mapboxLanguage);
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      this.context.map.removeControl(this.mapboxLanguage);
    }
  }, {
    key: 'render',
    value: function render() {
      return null;
    }
  }]);

  return MapboxLanguageControl;
}(_react.Component);

MapboxLanguageControl.contextTypes = {
  map: _propTypes2.default.object.isRequired
};

exports.default = MapboxLanguageControl;
